### DISCUSSION TITLE
how to use pmap in multi-host cpus

### DISCUSSION THREAD

**Discussion by JiahaoYao** [jax-ml/jax#10745] [Q&A] [answered]
I am able to run the multi-host cpu using jax. `pmap` and `jax.device_count` can not see the global devices? 

This [ONE](https://github.com/google/jax/blob/main/jax/_src/distributed.py#L35-L36) is not the solution?

**Comment by PhilipVinc** [comment, marked as answer]
Hi Jiahao

It's not an answer, but since I could not manage to use pmap at the time for that use case,  we wrote [mpi4jax](https://mpi4jax.readthedocs.io/en/latest/index.html) , so you just use standard jit and run the same code everywhere. It requires you to have a working MPI installation but I'm pretty sure at UC you have that...

Nowdays there's some experimental support through Global Sharded Devices Arrays but it's not greatly documented. I'll let other comment on how to do that...

**Reply by JiahaoYao** [reply]
thanks @PhilipVinc, appreciated your answers!
